### DISCUSSION TITLE
multi nodes multi gpu is support ？

### DISCUSSION THREAD

**Discussion by YLongJin** [bghira/SimpleTuner#516] [General]
multi nodes multi gpu is support ？

**Comment by bghira** [comment]
never tried.

**Comment by YLongJin** [comment]
thanks your reply

**Comment by YLongJin** [comment]
@bghira  can I train use the pixart dataset module？ the json file with image path and caption and prompt

**Comment by bghira** [comment]
possibly by just configuring the parquet backend to use it

**Comment by YLongJin** [comment]
the data is local, is this compatible？

**Comment by sayakpaul** [comment]
Local parquet loading shouldn't be an issue. If you face any, do report it.

**Comment by YLongJin** [comment]
ValueError: Your dataloader config must contain at least one image dataset AND at least one text_embed dataset. See this link for more information about dataset_type: https://github.com/bghira/SimpleTuner/blob/main/documentation/DATALOADER.md#configuration-options

**Comment by YLongJin** [comment]
I do not have text_embed dataset， i just have json file with image path and caption and prompt

**Comment by sayakpaul** [comment]
Oh then you should probably use the `datasets` library for that:

https://huggingface.co/docs/datasets/en/image_load

**Comment by YLongJin** [comment]
how can I make the embed dataset

**Comment by sayakpaul** [comment]
Cc: @bghira

**Comment by bghira** [comment]
please follow the quickstart and use the demo dataset first just so you know you have it working

**Comment by bghira** [comment]
the link to the quickstart is [here](https://github.com/bghira/SimpleTuner/blob/main/documentation/quickstart/SD3.md) for stable diffusion 3. I do not have a specific one for SDXL, but the basic concepts apply other than the difference for `MODEL_NAME` and `STABLE_DIFFUSION_3` variable properties.

here is what the config looks like:

```json
[
  {
    "id": "pseudo-camera-10k-sd3",
    "type": "local",
    "crop": true,
    "crop_aspect": "square",
    "crop_style": "center",
    "resolution": 0.5,
    "minimum_image_size": 0.25,
    "maximum_image_size": 1.0,
    "target_downsample_size": 1.0,
    "resolution_type": "area",
    "cache_dir_vae": "cache/vae/sd3/pseudo-camera-10k",
    "instance_data_dir": "datasets/pseudo-camera-10k",
    "disabled": false,
    "skip_file_discovery": "",
    "caption_strategy": "filename",
    "metadata_backend": "json"
  },
  {
    "id": "text-embeds",
    "type": "local",
    "dataset_type": "text_embeds",
    "default": true,
    "cache_dir": "cache/text/sd3/pseudo-camera-10k",
    "disabled": false,
    "write_batch_size": 128
  }
]
```

it's not anything you manually have to create - the trainer will do that for you.

this merely points to the storage location where these embeds can be stored.

it might seem needlessly complicated, and it is - because the trainer can split the storage locations of everything but the VAE cache objects, for efficiency purposes. you can store the image data locally via NVME and text embeds on a S3 storage bucket, for example.

hopefully after following the quickstart you'll have something working for your model, and then you can expand the configuration of `accelerate` to include multi-node.

because i don't have access to multiple nodes to train on, i have no ability to test or verify anything about configuration or runtime problems. please report any issues you do have, and we can work together on solving them.
